src/scipp/plt/model.py

```diff
@@ -0,0 +1,137 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+from .widgets.widget import WidgetView
+
+from typing import Tuple, Iterable
+from functools import partial
+
+
+class Node:
+
+    def __init__(self, func, name=None, views=None):
+        self.name = name
+        self.parent_name = None
+        self.dependency = None
+        self.func = func
+        self.views = list(views or [])
+
+    def send_notification(self, message):
+        for view in self.views:
+            view.notify({
+                "node_name": self.name,
+                "parent_name": self.parent_name,
+                "message": message
+            })
+
+    def request_data(self):
+        pipeline = []
+        node = self
+        while node.dependency is not None:
+            pipeline.append(node.func)
+            node = node.dependency
+        else:
+            pipeline.append(node.func)
+        pipeline = pipeline[::-1]
+        data = pipeline[0]()
+        for step in pipeline[1:]:
+            data = step(data)
+        return data
+
+    def add_view(self, view):
+        self.views.append(view)
+
+
+class Model:
+
+    def __init__(self, da):
+        self._name = da.name
```

Comment:
Motivated by the comment below: Data arrays usually have an empty name. It might be best to generate a new name here if `da.name == ''`. Or just call it 'data'?

Reply:
I think it's ok to have it as an empty string? It's the same as having it called `data`?

Reply:
I was concerned about getting conflicts if we have multiple data arrays. But we don't support that right now, so it should be fine.

Reply:
The current design implies multiple data arrays == multiple models.
As in the example with two 1d curves. I couldn't think of a case where we would need multiple data arrays inside a given model. But maybe you thought of something?

Reply:
I can't think of anything either. And I think that if we had such a case, we should have a dedicated model class that abstracts this away instead of putting multiple data arrays into the kind of model you wrote here.